pkg/policy/snapshots.go

```diff
@@ -0,0 +1,228 @@
+package policy
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/jeremywohl/flatten"
+)
+
+func (ce *Executor) StoreSnapshot(ctx context.Context, path string, tables []string) error {
+	if len(tables) == 0 {
+		return errors.New("no tables to snapshot")
+	}
+
+	for _, table := range tables {
+		ef, err := os.OpenFile(fmt.Sprintf("%s/table_%s.csv", path, table), os.O_CREATE|os.O_WRONLY, 0777)
+		if err != nil {
+			ce.log.Error("error opening file.", "error", err)
+			return err
+		}
+		defer ef.Close()
```

Comment:
This defer won't close files until all tables are read, which might lead to unnecessary number of open file handles. Would either make another fn that reads/processes a single file/table, or wrap it in a closure func so defer runs as soon as we're done with the file.

Reply:
I wrapped it in its own func